torchprime/torch_xla_models/train.py

```diff
@@ -330,11 +331,19 @@ def step_closure(epoch, step, loss, trace_start_time, trace_end_time):
 
       tpu_name = os.environ.get("TORCHPRIME_TPU_TYPE", None)
       if tpu_name:
-        # Add "torch_dtype" in model config
+        # Allow adding new keys to the config
+        OmegaConf.set_struct(self.config, False)
```

Comment:
Rather than trying to (IIUC) somewhat hackily stuff fields into the config, I think a cleaner approach could be the following:

- We actually add `torch_dtype` and `optimizer` to the `default.yaml` as real config fields
- In the trainer, check that `torch_dtype` is bf16 and `optimizer` is adafactor. That's the only supported ones for now. You can add a https://github.com/AI-Hypercomputer/torchprime/issues/36 bug reference for supporting AdamW.
- Once we do this, then we can naturally store those fields along with the rest of the config.
- This naturally prepares us to support other kind of dtype and optimizers in the future.

Reply:
Looks like latest PR moved the torch_dtype to default.yaml which LGTM